docs/DeveloperGuide.md

```diff
@@ -154,6 +154,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Sorting feature
+
+#### Implementation
+
+The sorting mechanism is facilitated by `SortCommand` and `SortCommandParser`. 
+Additionally, the mechanism utilises the following operations in `UniquePersonList`:
+
+* `UniquePersonList#sortByName(Boolean isReverse)` — Sorts the contact list by name in alphabetical order.
+* `UniquePersonList#sortByPhone(Boolean isReverse)` — Sorts the contact list by phone number in increasing order.
+* `UniquePersonList#sortByEmail(Boolean isReverse)` — Sorts the contact list by email in alphabetical order.
+* `UniquePersonList#sortByAddress(Boolean isReverse)` — Sorts the contact list by address in alphabetical order.
+* `UniquePersonList#sortByTag(Tag tag, Boolean isReverse)` — Sorts the contact list by a specified tag.
+
+These operations sort in reverse order when `isReverse` is true.
+
+These operations are exposed in the `Model` interface under the same method name.
+
+Given below is an example usage scenario and how the sorting mechanism behaves at each step.
+
+Step 1. The user executes `sort t/!friend n/` command to perform a multi-level sort. `SortCommandParser` calls `ArgumentTokenizer#tokenizeToList()` to separate the parameters of `t/!friend` and `n/`.
+
+Step 2. The `sort` command sorts the currently displayed list by name first, calling `Model#sortByName(Boolean isReverse)` where `isReverse = false`.
+
+Step 3. The `sort` command sorts the currently displayed list by the `friend` tag next, calling `Model#sortByTag(Tag tag, Boolean isReverse)` where `isReverse = true`.
+
+The following sequence diagram shows how the sort operation works:
+
+(insert sequence diagram here)
```

Comment:
For now this dummy placeholder is okay?

Reply:
I put a placeholder here to remind myself to insert a sequence diagram there in v1.3, hahaha. I can remove it, if you think it's out of place?

Reply:
Nope I think it's good!